Sources/ComposableArchitecture/Effects/Publisher.swift

```diff
@@ -21,7 +21,7 @@ public struct _EffectPublisher<Action>: Publisher {
   }
 
   public func receive(subscriber: some Combine.Subscriber<Action, Failure>) {
-    publisher.subscribe(subscriber)
+    publisher.receive(subscriber: subscriber)
   }
```

Comment:
I don't see why this change would fix the stack overflow, do you?

The docs for [`Publisher.subscribe(_:)`](https://developer.apple.com/documentation/combine/publisher/subscribe(_:)-4u8kn#discussion) read to me like the old implementation was right.